Eastern-Research-Group/mywaterway
# Feature/message alert banner

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3410978

## Main Changes:
* Add notification banner above app that reads data from a messages.json file in the server's public folder.
* Message is parsed as HTML using dangerouslySetInnerHTML 
* The "All" field in the JSON file applies a message to every page. If a specific page name is listed the banner will only be shown on that page.
* The Data and About pages will display the message of the page they were navigated from. This is acceptable behavior per a discussion with Brad Cooper because we will not need to display a specific message for these pages.
* No loading spinner is displayed while fetching. I am assuming the majority of the time no message will be shown in our app so adding a spinner that disappears and displays no content would be confusing to end users.
* If the fetch to the JSON file fails, no message is displayed for similar reasons. 
* Discussion about the loading/failure behavior is welcome.

## Steps To Test:
There are currently no messages to display so the file is empty {}
1. Check that no banner is displayed above the app while the JSON file is empty and no errors are in the console.
1. Replace the content of server/app/public/notifications/messages.json with:
```
{
  "all": {
    "color": "#721c24",
    "backgroundColor": "#f8d7da",
    "message": "<p>There will be <a href=\"https://www.epa.gov\" target=\"_blank\">scheduled maintenance</a> on the <strong>ATTAINS</strong> services on Thursday, July 16th starting at 8am and ending at 11am.</p>"
  },
  "community": {
    "color": "#0c5460",
    "backgroundColor": "#d1ecf1",
    "message": "<p>This is a message for the <strong>Community</strong> page. Check out <a href=\"epa.gov\">this link</a> for more information</p>"
  },
  "state": {
    "color": "#721c24",
    "backgroundColor": "#f8d7da",
    "message": "<p>State page message.</p>"
  },
  "plan-summary": {
    "color": "#721c24",
    "backgroundColor": "#f8d7da",
    "message": "Plan summary message."
  },
  "waterbody-report": {
    "color": "#721c24",
    "backgroundColor": "#f8d7da",
    "message": "Waterbody Report message."
  }
}
```
2. Reload the app and you should see 1 banner on the home page http://localhost:3000/
3. The link should be clickable and the word ATTAINS should be bold from a strong tag.
4. Navigate to the Community page http://localhost:3000/community
5. 2 banners should be displayed. 
6. Navigate to:
Plan Summary: http://localhost:3000/plan-summary/21AWIC/40958
State: http://localhost:3000/state
Waterbody Report: http://localhost:3000/waterbody-report/MDE_EASP/MD-02140204
And check that the pages have 2 banners.
7. Navigate to http://localhost:3000/national and check that only the "All" banner is displayed.
8. Block the proxy fetch to messages.json in the network tab and check that it retries 3 after failing.
9. Check that no error message or loading spinner is displayed when the fetch fails.
